api/src/main/java/org/openmrs/module/eptsreports/reporting/library/converter/CalculationResultDataConverter.java

```diff
@@ -0,0 +1,110 @@
+package org.openmrs.module.eptsreports.reporting.library.converter;
+
+import static org.openmrs.module.eptsreports.reporting.utils.EptsReportUtils.formatDate;
+import static org.openmrs.module.eptsreports.reporting.utils.EptsReportUtils.formatDateTime;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+import org.openmrs.Concept;
+import org.openmrs.Location;
+import org.openmrs.Patient;
+import org.openmrs.PatientState;
+import org.openmrs.PersonAttribute;
+import org.openmrs.calculation.result.CalculationResult;
+import org.openmrs.calculation.result.SimpleResult;
+import org.openmrs.module.reporting.data.converter.DataConverter;
+
+public class CalculationResultDataConverter implements DataConverter {
+
+  private String what;
+
+  public String getWhat() {
+    return what;
+  }
+
+  public void setWhat(String what) {
+    this.what = what;
+  }
+
+  public CalculationResultDataConverter() {}
+
+  public CalculationResultDataConverter(String what) {
+    this.what = what;
+  }
+
+  @Override
+  public Object convert(Object obj) {
+
+    if (obj == null) {
+      return "";
+    }
+
+    Object value = ((CalculationResult) obj).getValue();
+    if (value instanceof Boolean) {
```

Comment:
This is not a good design pattern since it is trying to pack too much diverse conditions into once class which breaks the single responsibility rule. Better we redesign this to use something like the strategy pattern or simply the whole thing down to SQL cohorts.

For example the deathDate of a patient - why not implement a SQL query as a cohort and display the result like that? Each time we need to add a new change (maybe a new column or a new data group, this approach means we need to edit this file to extend this if, else if... string of conditions. Generally that is a warning sign that we have broken the single responsibility rule.